src/tools/auth0/handlers/customDomains.ts

```diff
@@ -0,0 +1,90 @@
+import DefaultAPIHandler from './default';
+import { Asset, Assets } from '../../../types';
+
+export const schema = {
+  type: 'array',
+  items: {
+    type: 'object',
+    properties: {
+      custom_domain_id: { type: 'string' },
+      custom_client_ip_header: {
+        type: 'string',
+        nullable: true,
+        enum: ['true-client-ip', 'cf-connecting-ip', 'x-forwarded-for', null],
+      },
+      domain: { type: 'string' },
+      primary: { type: 'boolean' },
+      status: { type: 'string', enum: ['pending_verification', 'ready', 'disabled', 'pending'] },
+      type: { type: 'string', enum: ['auth0_managed_certs', 'self_managed_certs'] },
+      verifications: { type: 'object' },
+    },
+    required: ['domain', 'type'],
+  },
+};
+
+export default class CustomDomainsHadnler extends DefaultAPIHandler {
+  existing: Asset[] | null;
+
+  constructor(config: DefaultAPIHandler) {
+    super({
+      ...config,
+      type: 'customDomains',
+      id: 'custom_domain_id',
+      identifiers: ['domain'],
+      stripCreateFields: ['status', 'primary', 'verification'],
+      functions: {
+        //@ts-ignore
+        delete: (args) => {
+          return this.client.customDomains.delete({ id: args.custom_domain_id });
+        },
+      },
+    });
+  }
+
+  objString(item: Asset): string {
+    return super.objString(item.domain);
+  }
+
+  async getType(): Promise<Asset> {
+    if (this.existing) {
+      return this.existing;
+    }
+
+    const customDomains = await this.client.customDomains.getAll({ paginate: false });
```

Comment:
In general, yes, because those fields are still necessary for creating custom domains. This is necessary when going from tenant A to tenant B and wishing to have the same configuration for both. Although I do agree that it may not be apparent which fields are and aren't available for update. I wonder, does Terraform have a mechanism for making this clear to the user?